### PR TITLE
[32129] Add icon "Log time" close to spent time attribute in work packages details view

### PR DESCRIPTION
**Pull request by bsatarnejad**
### Todo

- [x] Show icon next to display field in work package view
  - [x] Do not show the icon, if the user does not have the permission to log time
- [x] A click on the icon, opens the time logging widget
   - [x] The work package is preselected (in the data)
   - [x] The work package select field should be hidden
- [x] When logging new time, the Work Package is updated accordingly
- [x] Add test

### CODE REVIEW

**Comment by bsatarnejad** [comment]
Ticket

https://community.openproject.com/projects/openproject/work_packages/32129/activity

File: frontend/src/app/modules/fields/display/field-types/wp-spent-time-display-field.module.ts [hunk 1]
```diff
@@ -65,5 +70,35 @@ export class WorkPackageSpentTimeDisplayField extends DurationDisplayField {
 
     element.innerHTML = '';
     element.appendChild(link);
+
+    this.appendTimelogLink(element);
+  }
+
+  private appendTimelogLink(element:HTMLElement) {
+    this
```
Comment (oliverguenther) [inline frontend/src/app/modules/fields/display/field-types/wp-spent-time-display-field.module.ts]: This method will result in 1 request per work package in the list and we should work on that before merging.

Looks like there is `workPackage.logTime` that should do the same thing?

**Review by oliverguenther — CHANGES_REQUESTED** [review]
LGTM functionality wise. I'd like to get rid of the extra request if possible. Looks like checking for the `workPackage.logTime` link is what we need.

File: frontend/src/app/modules/fields/display/field-types/wp-spent-time-display-field.module.ts (hunk 1 above)
Comment (HDinger) [reply frontend/src/app/modules/fields/display/field-types/wp-spent-time-display-field.module.ts]: Thanks! 👍 I missed that link.

**Pull request merged**